chainer/optimizers/ada_grad.py

```diff
@@ -39,13 +39,16 @@ class AdaGradRule(optimizer.UpdateRule):
     """
     _kernel = None
 
-    def __init__(self, parent_hyperparam=None, lr=None, eps=None):
+    def __init__(self, parent_hyperparam=None, lr=None, eps=None,
+                 batch_size_factor=None):
```

Comment:
`batch_size_factor` can be omitted? Is there any use case to set it in the constructor?

Reply:
The `test_hyperparams` method get the list of hyperparameters for the optimizer and tries to instantiate it passing all the hyperparameters as constructor arguments
```
        default = self.optimizer.hyperparam.get_dict()
        for name, default_value in six.iteritems(default):
            self.create()
            self.assertEqual(self.get_hyperparam(name), default_value)
            new_value = default_value + 0.1
            self.create(**{name: new_value})
            self.assertEqual(self.get_hyperparam(name), new_value)
```
Solutions for this are to explicitly disable the batch_size_factor in the test.
or support batch_size_factor hyperparameter in the optimizer constructor.

I am up for the first one.

Reply:
Just added the complete implementation of this proposal following the suggestions I got yesterday. Thanks!